src/main/scala/viper/gobra/frontend/ParseTreeTranslator.scala

```diff
@@ -0,0 +1,2240 @@
+  }
+
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitExprCaseClause(ctx: ExprCaseClauseContext): PExprSwitchCase = {
+    val left = visitExpressionList(ctx.exprSwitchCase().expressionList())
+    val body = PBlock(visitStatementList(ctx.statementList())).at(ctx)
+    PExprSwitchCase(left, body).at(ctx)
+  }
+
+  /** Visits a type switch node
+    *
+    * @param ctx the type switch context
+    * @return a positioned PTypeSwitch
+    */
+  override def visitTypeSwitchStmt(ctx: TypeSwitchStmtContext): PTypeSwitchStmt = {
+    val pre = visitNodeOrNone[PSimpleStmt](ctx.simpleStmt())
+    val binder = if (has(ctx.typeSwitchGuard().IDENTIFIER())) Some(idnDef.get(ctx.typeSwitchGuard().IDENTIFIER())) else None
+    val expr = visitNode[PExpression](ctx.typeSwitchGuard().primaryExpr())
+    // iterate through the cases and partition them into normal cases and the default case
+    val (dflt, cases) = ctx.typeCaseClause().asScala.toVector.partitionMap(clause =>
+      if (has(clause.typeSwitchCase().DEFAULT())) Left(visitTypeDefaultClause(clause).body) // default : <statements>
+      else Right(visitTypeCaseClause(clause)) // case <types> : <statements>
+    )
+    PTypeSwitchStmt(pre, expr, binder, cases, dflt).at(ctx)
+
+  }
+
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitTypeList(ctx: TypeListContext): Vector[PExpressionOrType] = {
+    val types = visitListNode[PType](ctx.type_())
+    // Need to check whether this includes nil, since it's a predeclared identifier and not a type
+    if (has(ctx.NIL_LIT()) && !ctx.NIL_LIT().isEmpty) types.appended(PNilLit().at(ctx.NIL_LIT(0))) else types
+  }
+
+  /**
+    * {@inheritDoc  }
+    *
+    * <p>The default implementation returns the result of calling
+    * {@link #   visitChildren} on {@code ctx}.</p>
+    */
+  override def visitTypeCaseClause(ctx: TypeCaseClauseContext): PTypeSwitchCase = {
+    val body = PBlock(visitStatementList(ctx.statementList()))
+      .at(if (has(ctx.statementList())) ctx.statementList() else ctx) // If we have no statement list, we need to position at the context
+    val left = visitTypeList(ctx.typeSwitchCase().typeList())
+    PTypeSwitchCase(left, body).at(ctx)
+  }
+
+  def visitTypeDefaultClause(ctx: TypeCaseClauseContext): PTypeSwitchDflt = {
+    val body = PBlock(visitStatementList(ctx.statementList())).at(ctx.statementList())
+    PTypeSwitchDflt(body).at(ctx)
+  }
+  //endregion
+
+  //region For statements
+  /**
+    * Visits a for statement with specifications
+    * @param specCtx
+    * @return a positioned PStatement
+    */
+  // TODO : Refactor
+  override def visitSpecForStmt(specCtx: SpecForStmtContext): PStatement = {
```

Comment:
Here it seems that you wanted to refactor something. Maybe use visitChildren instead of your current if-else cascade?

Reply:
ok.